src/orderly/cli/options.py

```diff
@@ -0,0 +1,130 @@
+import functools
+
+import click
+
+from outpack.search_options import SearchOptions
+
+
+class NumericalParamType(click.ParamType):
+    """
+    A type for numerical parameters.
+
+    It automatically deduces whether the parameter should be interpreted as an
+    int or a float.
+    """
+
+    name = "number"
+
+    def convert(self, value, param, ctx):
+        if isinstance(value, float):
+            return value
+
+        try:
+            return int(value)
+        except ValueError:
+            pass
+
+        try:
+            return float(value)
+        except ValueError:
+            self.fail(f"{value!r} is not a valid numerical value", param, ctx)
+
+
+# Almost all sub-commands accept a --root option, so it would make sense for it
```

Comment:
I'm still a bit on the fence about whether this is worth it. In my testing I was using the `--root` option a lot so this was quite nice. In practice though I'm not sure that many people will even use it.

Reply:
Git accepts `-C`, as does `make` with similar meaning - it's actually really useful in practice, and most users can ignore it

Reply:
Oh I definitely want to keep the option, the only thing I'm on the fence about is accepting it as `orderly --root /foo run ...`.

It's nice to work with interactively because I can go back it my shell history, remove the subcommand and replace it with another one, without having to type in the root again. But getting it to work involves a few hacks.

Interestingly, git's `-C` has slightly different semantics than the `--root` here. It is attached the the top level command, not each individual one (ie. `git commit -C path` doesn't work), and it works by changing the working directory before doing anything else. I think tar and make might be similar.

What this means is that any additional paths specified as arguments are now relative to the `-C` directory, not the directory the command was run from. It also means `git -C foo init` works, if foo already exists as a directory, and `git -C foo init bar` is equivalent to `git init foo/bar` (again assuming foo exists).